shell/common/shell.cc

```diff
@@ -522,7 +522,21 @@ Shell::~Shell() {
 
   vm_->GetServiceProtocol()->RemoveHandler(this);
 
-  fml::AutoResetWaitableEvent ui_latch, gpu_latch, platform_latch, io_latch;
+  fml::AutoResetWaitableEvent platiso_latch, ui_latch, gpu_latch,
+      platform_latch, io_latch;
+
+  fml::TaskRunner::RunNowOrPostTask(
+      task_runners_.GetPlatformTaskRunner(),
+      fml::MakeCopyable([this, &platiso_latch]() mutable {
+        const PlatformIsolateManager* platform_isolate_manager =
+            engine_->GetRuntimeController()->GetPlatformIsolateManager();
+        // PlatformIsolateManager::ShutdownPlatformIsolates is thread safe,
+        // so it's safe to const_cast here.
+        const_cast<PlatformIsolateManager*>(platform_isolate_manager)
```

Comment:
Add a ShutdownPlatformIsolates method to the `flutter::Engine` class that calls `runtime_controller_->ShutdownPlatformIsolates`

With that you can remove the const version of `PlatformConfigurationClient::GetShutdownIsolateManager` and avoid the const_cast here